javascript/code-challenges/challenges-05.test.js

```diff
@@ -0,0 +1,371 @@
+'use strict';
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 1 - Review
+
+Write a function that iterates over an array of people objects
+and creates a new list of each person's full name using the array method 'map'.
+Each object will have the shape {firstName:string, lastName:string}
+E.g. [ { firstName:"Jane", lastName:"Doe" }, { firstName:"James", lastName:"Bond"}]
+should convert to ["Jane Doe", "James Bond"]
+Note the space in between first and last names.
+You can assume that neither firstName nor lastName will be blank
+------------------------------------------------------------------------------------------------ */
+const toLastNames = people => {
+  let newarr = people.map( string => {
+    return (string.firstName + " " + string.lastName);   
+  });
+  return newarr;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 2
+
+Write a function named addValues that, given an array of numbers as input, uses reduce to add the values in the array.
+
+------------------------------------------------------------------------------------------------ */
+
+const addValues = (arr) => {
+  const addtogether = arr.reduce( (accumulator, val, idx) => {
+    return accumulator + val;
+  }, 0);
+  return addtogether;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 3
+
+Write a function named addPurchases that, given an array of objects as input, uses reduce to find the total amount purchased. Each object contains the keys `item` and `purchasePrice` like the example.
+
+{
+  item: 'switch'
+  purchasePrice: 399
+}
+
+------------------------------------------------------------------------------------------------ */
+
+const addPurchases = (arr) => {
+  const addtogether = arr.reduce( (accumulator, val, idx) => {
+    return accumulator + val.purchasePrice;
+  }, 0);
+  return addtogether;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 4
+
+Write a function named countNumberOfElements that, given an array as input, uses reduce to count the number of elements in the array.
+
+Note: You may not use the array's built-in length property.
+------------------------------------------------------------------------------------------------ */
+
+const countNumberOfElements = (arr) => {
+  const numchr = arr.reduce( (accumulator, val, idx) => {
+    return idx+1;
+  }, 0);
+  return numchr;
+};
+
```

Comment:
I think yours is the only solution I've seen that uses the third/currentIndex argument for reduce. Interesting. Was it just to get a hang of it or is that what occurred to your first when trying to solve the problem? Either way, well done.